codeql-config.yml

```diff
@@ -12,3 +12,8 @@ query-filters:
       id: js/missing-rate-limiting
   - exclude:
       id: js/missing-token-validation
+paths-ignore:
+  - '**/test*'
+  - '**/__test__/**'
+  - '**/__tests__/**'
+  - '**/*test.*'
```

Comment:
Hmm, looking at https://github.com/search?q=org%3Aopengovsg+path%3A*test*&type=code I see a couple of matches like:

- [testMode.tsx ](https://github.com/opengovsg/distributesg/blob/de659ba5a0ca83cb6f966a34f39f0b3a7bdb4aed/frontend/lib/helpers/testMode.tsx)
- [templates/testimonials.ts](https://github.com/opengovsg/ogp-templates/blob/f3c7437429215855012b11b88f0a5a91787af087/src/data/templates/testimonials.ts)

The rest that are in this search would be matched by the folder rules or maybe an additional folder:

```
- '**/test/*'
- '**/tests/*'
```

Would this make sense?

Reply:
Just to clarify is this what youre proposing?
```
- '**/*test*'
- '**/test/*'
- '**/tests/*'
```
I think that makes sense

Reply:
Ok I figured out a way to test across all repos and I tested with 
```
- '**/test/*'
- '**/tests/*'
- '**/__test__/*'
- '**/__tests__/*'
```

There are still some paths that are considered tests by AI and not covered

| Test File Type | Frequency | Pattern | Examples |
|----------------|-----------|---------|----------|
| Unit Tests | High | `*.test.ts`, `*.test.tsx`, `*.test.js` | [FormSG/frontend/src/features/admin-form/responses/FeedbackPage/utils/FeedbackCsvGenerator.test.ts](https://github.com/opengovsg/FormSG/blob/develop/frontend/src/features/admin-form/responses/FeedbackPage/utils/FeedbackCsvGenerator.test.ts)<br>[bright/src/pages/api/external/dev/sftp-aic360-test.ts](https://github.com/opengovsg/bright/blob/develop/src/pages/api/external/dev/sftp-aic360-test.ts) |
| End-to-End Tests | Medium | `*.e2e.test.ts`, `playwright/*.test.ts` | [bright/playwright/auth.e2e.test.ts](https://github.com/opengovsg/bright/blob/develop/playwright/auth.e2e.test.ts)<br>[isomer/apps/studio/playwright/smoke.test.ts](https://github.com/opengovsg/isomer/blob/develop/apps/studio/playwright/smoke.test.ts) |
| Configuration Files | Medium | `vitest.config.ts`, `jest-testcontainers-config.js` | [FormSG/frontend/vitest.config.ts](https://github.com/opengovsg/FormSG/blob/develop/frontend/vitest.config.ts)<br>[hms/packages/backend/jest-testcontainers-config.js](https://github.com/opengovsg/hms/blob/develop/packages/backend/jest-testcontainers-config.js) |
| CI/CD Test Workflows | Medium | `.github/workflows/*test*.yml` | [hms/.github/workflows/coverage_tests.yml](https://github.com/opengovsg/hms/blob/develop/.github/workflows/coverage_tests.yml)<br>[bright/.github/workflows/build_and_test.yml](https://github.com/opengovsg/bright/blob/develop/.github/workflows/build_and_test.yml) |
| Test Utilities | Low | `test-utils.tsx`, `testing.utils.ts` | [FormSG/frontend/src/test-utils.tsx](https://github.com/opengovsg/FormSG/blob/develop/frontend/src/test-utils.tsx)<br>[hms/packages/backend/src/modules/database/testing.utils.ts](https://github.com/opengovsg/hms/blob/develop/packages/backend/src/modules/database/testing.utils.ts) |
| Database Migrations for Tests | Low | `*-test-*.ts` (in migrations folders) | [hms/packages/backend/src/modules/database/migrations/1719079410866-test-result-updatedat.ts](https://github.com/opengovsg/hms/blob/develop/packages/backend/src/modules/database/migrations/1719079410866-test-result-updatedat.ts) |
| Load Tests | Low | `load-tests/*.test.ts` | [phas/load-tests/src/get-public-appointments.test.ts](https://github.com/opengovsg/phas/blob/develop/load-tests/src/get-public-appointments.test.ts)<br>[refer/load-test/src/pb.test.ts](https://github.com/opengovsg/refer/blob/develop/load-test/src/pb.test.ts) |
| Test Data/Fixtures | Very Low | `testing.fixtures.ts` | [hms/packages/backend/src/modules/database/testing.fixtures.ts](https://github.com/opengovsg/hms/blob/develop/packages/backend/src/modules/database/testing.fixtures.ts) |
| Mock Data | Very Low | `*.mock.ts` (inferred, not explicitly seen) | (No explicit examples found in the provided list) |
| Visual Regression Tests | Very Low | `visual-regression-tests.cy.js` | [askgov/cypress/e2e/visual-regression-tests.cy.js](https://github.com/opengovsg/askgov/blob/develop/cypress/e2e/visual-regression-tests.cy.js) |
| Smoke Tests | Very Low | `smoke.test.ts`, `smoke.e2e.test.ts` | [bright/playwright/smoke.e2e.test.ts](https://github.com/opengovsg/bright/blob/develop/playwright/smoke.e2e.test.ts)<br>[pixie/playwright/smoke.test.ts](https://github.com/opengovsg/pixie/blob/develop/playwright/smoke.test.ts) |

Of the above, I think we should add the unit tests and the playwright tests to the exclusion list with `*.test.*`

Reply:
Makes sense!